framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -418,16 +418,35 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            for (String expr : expressions) {
-                result.add(new Postcondition(expr, postcondAnno, anno));
+            try {
+                List<String> expressions =
+                        AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
+                AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+                if (postcondAnno == null) {
+                    continue;
+                }
+                for (String expr : expressions) {
+                    result.add(new Postcondition(expr, postcondAnno, anno));
+                }
+            } catch (Exception e) {
```

Comment:
It's a bit suspicious to catch *all* exceptions.  What is the specific case that you are checking for?  I hope that you can check it in a normal if statement instead, and simplify the logic.

Reply:
There would be many If statements i.e. for every annotation like 
```
if (type.equals("org.checkerframework.checker.nullness.qual.EnsuresNonNull")
                    || type.equals("org.checkerframework.checker.nullness.qual.EnsuresKeyFor")|| .....
```
It will look messy. Should I do it?

Reply:
Or I can catch a specific exception i.e ```ClassCastException```

Reply:
That would be better.  But the logic still seems strange.  It seems possible to iterate through most of the list, cause an exception, and then iterate through more of the list.  How about commenting what the exception means and how the second loop differs from the first one?

Reply:
I agree that this try/catch and the repeated logic is undesirable.
Why do you need it?
Don't these APIs look through the container API automatically and you only need to look for the single version?

Reply:
> I agree that this try/catch and the repeated logic is undesirable.
> Why do you need it?
> Don't these APIs look through the container API automatically and you only need to look for the single version?

Can you please elaborate? I am unable to understand.
If I will not edit this file, then it will give the error and It was originally working for the single version.